modal/stub.py

```diff
@@ -613,5 +616,49 @@ def _get_deduplicated_function_mounts(self, mounts: Dict[str, _Mount]):
             cached_mounts.append(self._function_mounts[root_path])
         return cached_mounts
 
+    async def spawn_sandbox(
+        self,
+        *entrypoint_args: str,
+        image: Optional[_Image] = None,  # The image to run as the container for the sandbox.
+        mounts: Sequence[_Mount] = (),  # Mounts to attach to the sandbox.
+        secrets: Sequence[_Secret] = (),  # Environment variables to inject into the sandbox.
+        network_file_systems: Dict[Union[str, os.PathLike], _NetworkFileSystem] = {},
+        timeout: Optional[int] = None,  # Maximum execution time of the sandbox in seconds.
+        workdir: Optional[str] = None,  # Working directory of the sandbox.
+        gpu: GPU_T = None,
+        cloud: Optional[str] = None,
+        cpu: Optional[float] = None,  # How many CPU cores to request. This is a soft limit.
+        memory: Optional[int] = None,  # How much memory to request, in MiB. This is a soft limit.
+    ) -> _Sandbox:
+        """Sandboxes are a way to run arbitrary commands in dynamically defined environments.
+
+        This function returns a [SandboxHandle](/docs/reference/modal.Sandbox#modalsandboxsandbox), which can be used to interact with the running sandbox.
+
+        Refer to the [docs](/docs/guide/sandbox) on how to spawn and use sandboxes.
+        """
+        from .sandbox import _Sandbox
+        from .stub import _default_image
+
+        if self._app is None:
+            raise InvalidError("`stub.spawn_sandbox` requires a running app.")
+
+        # TODO(erikbern): pulling a lot of app internals here, let's clean up shortly
+        resolver = Resolver(self._app._client, environment_name=self._app._environment_name, app_id=self._app._app_id)
```

Comment:
yeah this was the main reason for putting this on `App` earlier, but makes sense to do this now temporarily.

Reply:
Might make sense to merge the resolver and app classes later